### PR TITLE
feat: Convert nrg-checkbox to glimmer component

### PR DESCRIPTION
**Pull request by KeithClinard**

### CODE REVIEW

File: addon/components/nrg-checkbox.hbs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<div
+  class="ui checkbox
+    {{if @disabled "disabled"}}
+    {{if @fitted "fitted"}}
+    {{if @readonly "read-only"}}
+    {{@class}}"
+>
+  <Input
```
Comment (TSenter) [inline addon/components/nrg-checkbox.hbs:8]: I noticed for `<NrgTextField::TrimInput />` and #324 you used the basic HTML `<input />` and `<textarea />` tags, but here you're using Ember's `<Input />` component - what was the reason for that?

Comment (TSenter) [reply addon/components/nrg-checkbox.hbs:8]: If you use the Ember component `<Input />` I expect you'll need to prefix the properties with an `@`, eg.
```js
  @id={{this.focusId}}
  @type="checkbox"
  @checked={{this.checked}}
```

File: addon/components/nrg-checkbox.js [hunk 2]
```diff
@@ -1,31 +1,18 @@
...
+export default class NrgCheckboxComponent extends NrgValidationComponent {
+  @tracked
+  focusId = 'field-' + guidFor(this);
 
-  checked: alias('value'),
-  _checked: bool('checked'),
-  bindValue: true,
+  get checked() {
+    return !!this.value;
+  }
```
Comment (TSenter) [inline addon/components/nrg-checkbox.js:12]: Can you use the `@bool` computed property here?
```suggestion
  @bool('value')
  checked;
```

File: tests/dummy/app/view-components/nrg-forms/template.hbs [hunk 3]
```diff
@@ -69,7 +69,7 @@
         <field.checkbox @label="Inline Checkbox" @model={{this}} @valuePath="checkbox1" @disabled={{disabled}} />
       </form.field>
       <form.field @required={{required}} as |field|>
-        <field.checkbox @type="slider" @label="Inline Slider" @checked={{true}} @disabled={{disabled}} />
+        <field.checkbox @type="slider" @label="Inline Slider" @disabled={{disabled}} />
```
Comment (TSenter) [inline tests/dummy/app/view-components/nrg-forms/template.hbs:72]: Does this need a model and valuePath?

**Review by TSenter — CHANGES_REQUESTED** [review]

File: addon/components/nrg-checkbox.js (hunk 2 above)
Comment (KeithClinard) [reply addon/components/nrg-checkbox.js:12]: Using a computed property results in this linting error:
image

File: tests/dummy/app/view-components/nrg-forms/template.hbs (hunk 3 above)
Comment (KeithClinard) [reply tests/dummy/app/view-components/nrg-forms/template.hbs:72]: If I wanted to do anything important with it, yes.
In this case it's only there to demonstrate the style variations of the checkbox, so it's value is unbound.

File: addon/components/nrg-checkbox.hbs (hunk 1 above)
Comment (KeithClinard) [reply addon/components/nrg-checkbox.hbs]: Good catch. This was not intended. I'm trying to use raw html tags wherever possible.

File: addon/components/nrg-checkbox.js (hunk 2 above)
Comment (TSenter) [reply addon/components/nrg-checkbox.js:12]: That's very strange... the [official API docs](https://api.emberjs.com/ember/3.28/functions/@ember%2Fobject%2Fcomputed/bool) show the decorator version for a native class, but the [eslint rule](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-computed-properties-in-native-classes.md) says to not use them. In that case, I'm good with your version.

**Review by TSenter — APPROVED** [review]

**Pull request merged**